projects/custody-key-dates-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/messaging/Handler.kt

```diff
@@ -24,11 +24,18 @@ class Handler(
                 ?.let { cduService.updateCustodyKeyDates(it) }
 
             is CustodyDateChanged -> cduService.updateCustodyKeyDates(message.bookingId)
+            is ProbationOffenderEvent -> when (notification.eventType) {
+                "SENTENCE_CHANGED",
+                -> cduService.updateCustodyKeyDates(cduService.findNomsByCrn(message.crn))
+
+                else -> throw IllegalArgumentException("Unexpected offender event type: ${notification.eventType}")
+            }
         }
     }
 }
 
 data class CustodyDateChanged(val bookingId: Long)
+data class ProbationOffenderEvent(val crn: String)
```

Comment:
Could we annotate these with the `@Message` class.  And also add the new event type to the `@Publish` annotation on the handler method - so it updates the AsyncAPI doc.

You might need to merge from main to get the AsyncAPI stuff.

Reply:
ok ill rebase

Reply:
done